src/vs/workbench/contrib/notebook/browser/notebookExecutionStateServiceImpl.ts

```diff
@@ -139,11 +140,45 @@ export class NotebookExecutionStateService extends Disposable implements INotebo
 			exe.onDidComplete(lastRunSuccess => this._onCellExecutionDidComplete(notebookUri, cellHandle, exe, lastRunSuccess)));
 		this._cellListeners.set(CellUri.generate(notebookUri, cellHandle), disposable);
 
+		notebook.onWillAddRemoveCells((e: NotebookTextModelWillAddRemoveEvent) => {
+			const lastFailedCell = this.getLastFailedCellForNotebook(notebook.uri);
+			e.rawEvent.changes.forEach(([start, deleteCount]) => {
+				if (deleteCount) {
+					const deletedHandles = notebook.cells.slice(start, start + deleteCount).map(c => c.handle);
```

Comment:
We should probably check the index of `lastFailedCell` and then compare it against `[start, start + deletedCount]`.